### PR TITLE
Taskloop & taskloop SIMD w/ shared clause

### PR DESCRIPTION
**Pull request by nolanbaker31**
Both tests pass on LLVM 17, but the SIMD test has error message 

`loop not vectorized: the optimizer was unable to perform the requested transformation; the transformation might be disabled or specified as part of an unsupported transformation ordering`

### CODE REVIEW

File: tests/4.5/taskloop/test_taskloop_simd_shared.c [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+  for(int i = 0; i < N; i++){
+    A[i] = 1;
+    B[i] = 2;
+    C[i] = 3;
+  }
+
+#pragma omp parallel
+  {
+    int s_val=4;
+
```
Comment (nolanbaker31) [inline tests/4.5/taskloop/test_taskloop_simd_shared.c:31]: add a single

File: tests/4.5/taskloop/test_taskloop_shared.c [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+
+int taskloop_shared() {
+
+  int errors = 0;
+  int i, j;
+
+#pragma omp parallel
+  {
+    int s_val=0;
+
```
Comment (nolanbaker31) [inline tests/4.5/taskloop/test_taskloop_shared.c:25]: add single

File: tests/4.5/taskloop/test_taskloop_shared.c [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+  int errors = 0;
+  int s_val=0;
+  int check = -1;
+
+  #pragma omp target map(tofrom: check, s_val)//map(tofrom: s_val) map(from: check)
+  {
+    check = 7;
+    #pragma omp taskloop shared(s_val)
+      for (int i = 0; i < N; ++i){
+        ++s_val;
```
Comment (tob2) [inline tests/4.5/taskloop/test_taskloop_shared.c:27]: This looks unsafe. Don't you need something like 'omp atomic update' — or 'omp taskloop reduction(+:s_val)' here?

**Comment by andrewkallai** [comment]
frontier
clang 18.0.0
test_taskloop_shared.c.o:
[OMPVV_ERROR: test_taskloop_shared.c:29] Value of s_val should be 1024, received 0
[OMPVV_INFO: test_taskloop_shared.c:36] Test is running on device.
[OMPVV_INFO: test_taskloop_shared.c:38] The value of errors is 1.
[OMPVV_RESULT: test_taskloop_shared.c] Test failed on the device.

File: tests/4.5/taskloop/test_taskloop_shared.c [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+
+#define N 1024
+
+int taskloop_shared() {
+
+  int errors = 0;
+  int s_val=0;
+
+  #pragma omp target map(tofrom: s_val)
+  {
```
Comment (spophale) [inline tests/4.5/taskloop/test_taskloop_shared.c:22]: #pragma amp parallel
#pragma amp single

**Review by fel-cab — APPROVED** [review]

**Comment by fel-cab** [comment]
The test passes with LLVM on Perlmutter, but not on Frontier, so I think the test is correct.

**Review by seyonglee — APPROVED** [review]
It looks good to me.
[Results on Summit]
- test_taskloop_shared.c
    - GCC 13.1.1: C test passed.
    - XL 16.1.1-10: C test passed but ran on the host 
    - NVHPC 22.11: line 23: error: this OpenMP construct is not supported in NVIDIA subset: The 'taskloop' construct is not supported.
    - LLVM 17.0.0: C test passed.
- test_taskloop_simd_shared.c
    - GCC 13.1.1: C test passed.
    - XL 16.1.1-10: C test passed.
    - NVHPC 22.11: line 32: error: this OpenMP construct is not supported in NVIDIA subset: The 'taskloop' construct is not supported.
    - LLVM 17.0.0: C test passed.

**Pull request merged**